internal/blueprint/filesystem_customizations_test.go

```diff
@@ -0,0 +1,173 @@
+package blueprint
+
+import (
+	"testing"
+
+	"github.com/BurntSushi/toml"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetFilesystems(t *testing.T) {
+
+	expectedFilesystems := []FilesystemCustomization{
+		{
+			MinSize:    1024,
+			Mountpoint: "/",
+		},
+	}
+
+	TestCustomizations := Customizations{
+		Filesystem: expectedFilesystems,
+	}
+
+	retFilesystems := TestCustomizations.GetFilesystems()
+
+	assert.ElementsMatch(t, expectedFilesystems, retFilesystems)
+}
+
+func TestGetFilesystemsMinSize(t *testing.T) {
+
+	expectedFilesystems := []FilesystemCustomization{
+		{
+			MinSize:    1024,
+			Mountpoint: "/",
+		},
+		{
+			MinSize:    4096,
+			Mountpoint: "/var",
+		},
+	}
+
+	TestCustomizations := Customizations{
+		Filesystem: expectedFilesystems,
+	}
+
+	retFilesystemsSize := TestCustomizations.GetFilesystemsMinSize()
+
+	assert.EqualValues(t, uint64(5120), retFilesystemsSize)
+}
+
+func TestGetFilesystemsMinSizeNonSectorSize(t *testing.T) {
+
+	expectedFilesystems := []FilesystemCustomization{
+		{
+			MinSize:    1025,
+			Mountpoint: "/",
+		},
+		{
+			MinSize:    4097,
+			Mountpoint: "/var",
+		},
+	}
+
+	TestCustomizations := Customizations{
+		Filesystem: expectedFilesystems,
+	}
+
+	retFilesystemsSize := TestCustomizations.GetFilesystemsMinSize()
+
+	assert.EqualValues(t, uint64(5632), retFilesystemsSize)
+}
+
+var tomlSizeSet = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+size = 1024
+`
+
+var tomlMinSizeSet = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+minsize = 1024
+`
+
+var tomlSizeAndMinSizeSetAndEqual = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+size = 1024
+minsize = 1024
+`
+
+var tomlSizeAndMinSizeSetAndSizeSmaller = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+size = 1024
+minsize = 2048
+`
+
+var tomlSizeAndMinSizeSetAndMinSizeSmaller = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+size = 2048
+minsize = 1024
+`
+
+var tomlSizeAndMinSizeNotSet = `
+[[customizations.filesystem]]
+mountpoint = "/var"
+`
```

Comment:
I would find it nicer if you just in-lined these in the `TestGetFilesystemsMinSizeTOML()` test cases. Also, please add a case when the size is specified as a string, which would reveal the c&p error that I found out in the TOML unmarshaling code.

Reply:
Sure